# Table of Contents

0.1 Importing Libraries

0.2 Importing Data

0.3 Exploring Original Df

0.4 Create new columns with profile flags

    0.4.1  total_items_sold_by_dept
    
    0.4.2  total_spent_in_dept
    
    0.4.3  avg_price_per_item_in_dept
    
    0.4.4  total_orders_from_dept
    
    0.4.5  avg_items_per_order_by_dept
    
    0.4.6  n_unique_items_in_dept 
      

0.5 Confirming Final Df 

0.6 Exporting Final Df

### 0.1 Importing Libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

### 0.2 Importing Data

In [2]:
# Identify the file pathway to data files
path = r'C:\Users\CJ\Documents\_CJ-Stuff\Career Foundry\Data Immersion\Ach 4 - Python\2023-03 Instacart Basket Analysis'

In [3]:
# Import data
df = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_all3.pkl'))

In [4]:
dept_data = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'department_data.csv'), index_col = 0)

### 0.3 Exploring Original Dfs

In [5]:
df.shape

(32434489, 45)

In [6]:
dept_data.shape

(21, 1)

In [7]:
df.head()

order_id  user_id  order_number  order_day_of_week  order_hour_of_day  \
0   2539329        1             1                  2                  8   
1   2539329        1             1                  2                  8   
2   2539329        1             1                  2                  8   
3   2539329        1             1                  2                  8   
4   2539329        1             1                  2                  8   

   days_since_prior_order  first_order_of_new_customer  product_id  \
0                     NaN                         True         196   
1                     NaN                         True       14084   
2                     NaN                         True       12427   
3                     NaN                         True       26088   
4                     NaN                         True       26405   

   add_to_cart_order  reordered  ... total_spent_in_order  \
0                  1          0  ...                 31.6   
1                  2          0  ...                 31.6   
2                  3          0  ...                 31.6   
3                  4          0  ...                 31.6   
4                  5          0  ...                 31.6   

   total_spent_in_lifetime  total_items_in_lifetime  \
0               375.700012                       59   
1               375.700012                       59   
2               375.700012                       59   
3               375.700012                       59   
4               375.700012                       59   

   avg_items_per_order_by_user  baby_parent  pet_owner income_group  \
0                     5.898438        False      False   Low income   
1                     5.898438        False      False   Low income   
2                     5.898438        False      False   Low income   
3                     5.898438        False      False   Low income   
4                     5.898438        False      False   Low income   

  senior_citizen two_parent_fam single_parent  
0          False           True         False  
1          False           True         False  
2          False           True         False  
3          False           True         False  
4          False           True         False  

[5 rows x 45 columns]

In [8]:
dept_data.head(10)

department
1            frozen
2             other
3            bakery
4           produce
5           alcohol
6     international
7         beverages
8              pets
9   dry goods pasta
10             bulk

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 32434489 entries, 0 to 32434488
Data columns (total 45 columns):
 #   Column                       Dtype         
---  ------                       -----         
 0   order_id                     int64         
 1   user_id                      int64         
 2   order_number                 int16         
 3   order_day_of_week            int8          
 4   order_hour_of_day            int8          
 5   days_since_prior_order       float16       
 6   first_order_of_new_customer  bool          
 7   product_id                   int32         
 8   add_to_cart_order            int16         
 9   reordered                    int8          
 10  product_name                 object        
 11  aisle_id                     float16       
 12  department_id                float16       
 13  price                        float32       
 14  product_info_available       bool          
 15  price_available              bool          
 16

In [10]:
dept_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21 entries, 1 to 21
Data columns (total 1 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   department  21 non-null     object
dtypes: object(1)
memory usage: 336.0+ bytes


In [11]:
# Checking for nulls
df.isnull().sum()

order_id                             0
user_id                              0
order_number                         0
order_day_of_week                    0
order_hour_of_day                    0
days_since_prior_order         2078068
first_order_of_new_customer          0
product_id                           0
add_to_cart_order                    0
reordered                            0
product_name                     30770
aisle_id                         30770
department_id                    30770
price                            35897
product_info_available               0
price_available                      0
price_range                      35897
busiest_day                          0
busiest_days                         0
busiest_period_of_day                0
max_order                            0
loyalty_flag                         0
mean_spending                        0
spender_type                         0
median_days_since_ordering           0
frequent_shopper         

In [12]:
# The null values for days_since_prior_order 
# are flagged by first_order of new customer
df['first_order_of_new_customer'].value_counts()

False    30356421
True      2078068
Name: first_order_of_new_customer, dtype: int64

In [13]:
# The null values for price
# are flagged by price_available
df['price_available'].value_counts()

True     32398592
False       35897
Name: price_available, dtype: int64

In [14]:
# The null values for product_name, aisle_id, and department_id 
# are flagged by product_info_available
df['product_info_available'].value_counts()

True     32403719
False       30770
Name: product_info_available, dtype: int64

All null/NaN values are appropriately flagged.

## 0.4  Create new columns with profile flags

In [15]:
# Identifying the number of unique department_id numbers
df.department_id.nunique()

21

In [16]:
# Add a 'department_id' column for ease of merging later on
dept_data['department_id'] = dept_data.index

In [17]:
dept_data

department  department_id
1            frozen              1
2             other              2
3            bakery              3
4           produce              4
5           alcohol              5
6     international              6
7         beverages              7
8              pets              8
9   dry goods pasta              9
10             bulk             10
11    personal care             11
12     meat seafood             12
13           pantry             13
14        breakfast             14
15     canned goods             15
16       dairy eggs             16
17        household             17
18           babies             18
19           snacks             19
20             deli             20
21          missing             21

###     0.4.1  total_items_sold_by_dept

In [18]:
# Preview the number of items sold in each department
df['department_id'].value_counts(dropna=False).sort_index()

1.0     2234743
2.0       36291
3.0     1172428
4.0     9479291
5.0      153696
6.0      269253
7.0     2688115
8.0       97716
9.0      866627
10.0      34573
11.0     447010
12.0     708927
13.0    1875369
14.0     703033
15.0    1068058
16.0    5398185
17.0     738666
18.0     423794
19.0    2887550
20.0    1051249
21.0      69145
NaN       30770
Name: department_id, dtype: int64

Note there are 30,770 items where no product info is available.
This is in addition to the 69,145 items sold without a department name ('Missing')

In [19]:
# Create a column in the dept_data df to hold these counts.
dept_data['total_items_sold_by_dept'] = df['department_id'].value_counts().sort_index()

In [20]:
dept_data

department  department_id  total_items_sold_by_dept
1            frozen              1                   2234743
2             other              2                     36291
3            bakery              3                   1172428
4           produce              4                   9479291
5           alcohol              5                    153696
6     international              6                    269253
7         beverages              7                   2688115
8              pets              8                     97716
9   dry goods pasta              9                    866627
10             bulk             10                     34573
11    personal care             11                    447010
12     meat seafood             12                    708927
13           pantry             13                   1875369
14        breakfast             14                    703033
15     canned goods             15                   1068058
16       dairy eggs             16                   5398185
17        household             17                    738666
18           babies             18                    423794
19           snacks             19                   2887550
20             deli             20                   1051249
21          missing             21                     69145

In [21]:
dept_data.shape

(21, 3)

###  0.4.2  total_spent_in_dept

In [22]:
# Checking the total spent across the df
df['price'].sum()

252418420.0

In [23]:
# Create a column showing the total spent in each department
df['total_spent_in_dept'] = df.groupby(['department_id'])['price'].transform('sum')

In [24]:
# Confirming the column was created correctly
df[['department_id', 'price', 'total_spent_in_dept']].head()

department_id  price  total_spent_in_dept
0            7.0    9.0           20646084.0
1           16.0   12.5           44954496.0
2           19.0    4.4           12345538.0
3           19.0    4.7           12345538.0
4           17.0    1.0            5454352.5

In [25]:
# Create a subset of the df where each row = 1 department
# insted of each row being 1 item in an order

# Empty df
df_subset = pd.DataFrame()

In [26]:
# Subset only necessary columns: user_id, order_id, and days_since_prior
df_subset = df[['department_id', 'total_spent_in_dept']]

In [27]:
# Checking the subset
df_subset.head()

department_id  total_spent_in_dept
0            7.0           20646084.0
1           16.0           44954496.0
2           19.0           12345538.0
3           19.0           12345538.0
4           17.0            5454352.5

In [28]:
# Remove duplicate rows
df_subset = df_subset.drop_duplicates()

In [29]:
# Setting up display option to stay in decimal form rather than scientific notation
pd.set_option('display.float_format', lambda x: f'{x:,.3f}')

In [30]:
# Checking resulting df
df_subset.value_counts(dropna=False).sort_index()

department_id  total_spent_in_dept
1.000          17,289,206.000         1
2.000          253,708.000            1
3.000          9,208,026.000          1
4.000          75,660,936.000         1
5.000          1,251,654.250          1
6.000          2,068,472.750          1
7.000          20,646,084.000         1
8.000          770,888.375            1
9.000          6,369,953.500          1
10.000         288,648.594            1
11.000         3,573,172.750          1
12.000         11,551,570.000         1
13.000         15,020,661.000         1
14.000         5,644,626.000          1
15.000         8,053,132.000          1
16.000         44,954,496.000         1
17.000         5,454,352.500          1
18.000         3,237,048.500          1
19.000         12,345,538.000         1
20.000         8,177,352.000          1
21.000         598,845.188            1
NaN            NaN                    1
dtype: int64

The sum of the the department totals is ~22,418,371.41, which is ~48.59 less than the sum across the whole df.  While this is not a perfet match, it is an percent error rate of < 0.00000019% and thus acceptable for our purposes.

And we're not worried about the NaN values since they'll drop away when we do a left merge.

In [31]:
# Merge the newly derived totals for each department with the dept_data df
dept_data = dept_data.merge(df_subset, on = ['department_id'], how = 'left')

In [32]:
dept_data

department  department_id  total_items_sold_by_dept  \
0            frozen              1                   2234743   
1             other              2                     36291   
2            bakery              3                   1172428   
3           produce              4                   9479291   
4           alcohol              5                    153696   
5     international              6                    269253   
6         beverages              7                   2688115   
7              pets              8                     97716   
8   dry goods pasta              9                    866627   
9              bulk             10                     34573   
10    personal care             11                    447010   
11     meat seafood             12                    708927   
12           pantry             13                   1875369   
13        breakfast             14                    703033   
14     canned goods             15                   1068058   
15       dairy eggs             16                   5398185   
16        household             17                    738666   
17           babies             18                    423794   
18           snacks             19                   2887550   
19             deli             20                   1051249   
20          missing             21                     69145   

    total_spent_in_dept  
0        17,289,206.000  
1           253,708.000  
2         9,208,026.000  
3        75,660,936.000  
4         1,251,654.250  
5         2,068,472.750  
6        20,646,084.000  
7           770,888.375  
8         6,369,953.500  
9           288,648.594  
10        3,573,172.750  
11       11,551,570.000  
12       15,020,661.000  
13        5,644,626.000  
14        8,053,132.000  
15       44,954,496.000  
16        5,454,352.500  
17        3,237,048.500  
18       12,345,538.000  
19        8,177,352.000  
20          598,845.188

In [33]:
# Cleaning up memory
del df_subset

In [34]:
dept_data.shape

(21, 4)

###  0.4.3  avg_price_per_item_in_dept

In [35]:
# The avg_price_per_item_in_dept = total_spent_in_dept 
# divided by total_items_sold_in_dept
dept_data['avg_price_per_item_in_dept'] = dept_data['total_spent_in_dept']/dept_data['total_items_sold_by_dept']

In [36]:
dept_data

department  department_id  total_items_sold_by_dept  \
0            frozen              1                   2234743   
1             other              2                     36291   
2            bakery              3                   1172428   
3           produce              4                   9479291   
4           alcohol              5                    153696   
5     international              6                    269253   
6         beverages              7                   2688115   
7              pets              8                     97716   
8   dry goods pasta              9                    866627   
9              bulk             10                     34573   
10    personal care             11                    447010   
11     meat seafood             12                    708927   
12           pantry             13                   1875369   
13        breakfast             14                    703033   
14     canned goods             15                   1068058   
15       dairy eggs             16                   5398185   
16        household             17                    738666   
17           babies             18                    423794   
18           snacks             19                   2887550   
19             deli             20                   1051249   
20          missing             21                     69145   

    total_spent_in_dept  avg_price_per_item_in_dept  
0        17,289,206.000                       7.737  
1           253,708.000                       6.991  
2         9,208,026.000                       7.854  
3        75,660,936.000                       7.982  
4         1,251,654.250                       8.144  
5         2,068,472.750                       7.682  
6        20,646,084.000                       7.681  
7           770,888.375                       7.889  
8         6,369,953.500                       7.350  
9           288,648.594                       8.349  
10        3,573,172.750                       7.993  
11       11,551,570.000                      16.294  
12       15,020,661.000                       8.009  
13        5,644,626.000                       8.029  
14        8,053,132.000                       7.540  
15       44,954,496.000                       8.328  
16        5,454,352.500                       7.384  
17        3,237,048.500                       7.638  
18       12,345,538.000                       4.275  
19        8,177,352.000                       7.779  
20          598,845.188                       8.661

In [37]:
dept_data.shape

(21, 5)

### 0.4.4 total_orders_from_dept

In [38]:
# Create a tempoary df to hold the count of the total_orders_from_dept
count_of_orders = pd.DataFrame()

In [39]:
count_of_orders['total_orders_from_dept'] = df.groupby('department_id').order_id.nunique()

In [40]:
# Checking temp df
count_of_orders

total_orders_from_dept
department_id                        
1.000                         1180461
2.000                           35056
3.000                          878658
4.000                         2409320
5.000                           84689
6.000                          221537
7.000                         1456996
8.000                           59281
9.000                          597862
10.000                          33802
11.000                         318498
12.000                         574729
13.000                        1117763
14.000                         521401
15.000                         681305
16.000                        2174739
17.000                         470780
18.000                         177710
19.000                        1391447
20.000                         770300
21.000                          59477

In [41]:
count_of_orders.shape

(21, 1)

In [42]:
# Merging count_of_orders with dept_data
dept_data = dept_data.merge(count_of_orders, on = ['department_id'], how = 'left', indicator = True)

In [43]:
# Checking the merge
dept_data

department  department_id  total_items_sold_by_dept  \
0            frozen              1                   2234743   
1             other              2                     36291   
2            bakery              3                   1172428   
3           produce              4                   9479291   
4           alcohol              5                    153696   
5     international              6                    269253   
6         beverages              7                   2688115   
7              pets              8                     97716   
8   dry goods pasta              9                    866627   
9              bulk             10                     34573   
10    personal care             11                    447010   
11     meat seafood             12                    708927   
12           pantry             13                   1875369   
13        breakfast             14                    703033   
14     canned goods             15                   1068058   
15       dairy eggs             16                   5398185   
16        household             17                    738666   
17           babies             18                    423794   
18           snacks             19                   2887550   
19             deli             20                   1051249   
20          missing             21                     69145   

    total_spent_in_dept  avg_price_per_item_in_dept  total_orders_from_dept  \
0        17,289,206.000                       7.737                 1180461   
1           253,708.000                       6.991                   35056   
2         9,208,026.000                       7.854                  878658   
3        75,660,936.000                       7.982                 2409320   
4         1,251,654.250                       8.144                   84689   
5         2,068,472.750                       7.682                  221537   
6        20,646,084.000                       7.681                 1456996   
7           770,888.375                       7.889                   59281   
8         6,369,953.500                       7.350                  597862   
9           288,648.594                       8.349                   33802   
10        3,573,172.750                       7.993                  318498   
11       11,551,570.000                      16.294                  574729   
12       15,020,661.000                       8.009                 1117763   
13        5,644,626.000                       8.029                  521401   
14        8,053,132.000                       7.540                  681305   
15       44,954,496.000                       8.328                 2174739   
16        5,454,352.500                       7.384                  470780   
17        3,237,048.500                       7.638                  177710   
18       12,345,538.000                       4.275                 1391447   
19        8,177,352.000                       7.779                  770300   
20          598,845.188                       8.661                   59477   

   _merge  
0    both  
1    both  
2    both  
3    both  
4    both  
5    both  
6    both  
7    both  
8    both  
9    both  
10   both  
11   both  
12   both  
13   both  
14   both  
15   both  
16   both  
17   both  
18   both  
19   both  
20   both

In [44]:
dept_data['_merge'].value_counts()

both          21
left_only      0
right_only     0
Name: _merge, dtype: int64

In [45]:
# Removing the merge flag so it doesn't interfere with future merges
dept_data = dept_data.drop(columns = ['_merge'])

In [46]:
dept_data.head()

department  department_id  total_items_sold_by_dept  total_spent_in_dept  \
0     frozen              1                   2234743       17,289,206.000   
1      other              2                     36291          253,708.000   
2     bakery              3                   1172428        9,208,026.000   
3    produce              4                   9479291       75,660,936.000   
4    alcohol              5                    153696        1,251,654.250   

   avg_price_per_item_in_dept  total_orders_from_dept  
0                       7.737                 1180461  
1                       6.991                   35056  
2                       7.854                  878658  
3                       7.982                 2409320  
4                       8.144                   84689

In [47]:
dept_data.shape

(21, 6)

In [48]:
# Reclaiming space
del count_of_orders

### 0.4.5  avg_items_per_order_by_dept

In [49]:
# Create a column with the avg_items_per_order_by_dept
dept_data['avg_items_per_order_by_dept'] = dept_data['total_items_sold_by_dept'] / dept_data['total_orders_from_dept']

In [50]:
# Exploring new column
dept_data[['department', 'avg_items_per_order_by_dept']]

department  avg_items_per_order_by_dept
0            frozen                        1.893
1             other                        1.035
2            bakery                        1.334
3           produce                        3.934
4           alcohol                        1.815
5     international                        1.215
6         beverages                        1.845
7              pets                        1.648
8   dry goods pasta                        1.450
9              bulk                        1.023
10    personal care                        1.403
11     meat seafood                        1.233
12           pantry                        1.678
13        breakfast                        1.348
14     canned goods                        1.568
15       dairy eggs                        2.482
16        household                        1.569
17           babies                        2.385
18           snacks                        2.075
19             deli                        1.365
20          missing                        1.163

In [51]:
dept_data['avg_items_per_order_by_dept'].describe()

count   21.000
mean     1.689
std      0.651
min      1.023
25%      1.334
50%      1.568
75%      1.845
max      3.934
Name: avg_items_per_order_by_dept, dtype: float64

In [52]:
dept_data.shape

(21, 7)

### 0.4.6  n_unique_items_in_dept

In [53]:
# Create a tempoary df to hold the count of the n_unique_items_in_dept
temp = pd.DataFrame()

In [54]:
temp['n_unique_items_in_dept'] = df.groupby('department_id').product_id.nunique()

In [55]:
temp

n_unique_items_in_dept
department_id                        
1.000                            4006
2.000                             548
3.000                            1515
4.000                            1684
5.000                            1054
6.000                            1139
7.000                            4360
8.000                             971
9.000                            1858
10.000                             38
11.000                           6559
12.000                            906
13.000                           5368
14.000                           1113
15.000                           2092
16.000                           3443
17.000                           3083
18.000                           1080
19.000                           6263
20.000                           1322
21.000                           1255

In [56]:
# Merging count_of_orders with dept_data
dept_data = dept_data.merge(temp, on = ['department_id'], how = 'left', indicator = True)

In [57]:
# Checking the merge
dept_data

department  department_id  total_items_sold_by_dept  \
0            frozen              1                   2234743   
1             other              2                     36291   
2            bakery              3                   1172428   
3           produce              4                   9479291   
4           alcohol              5                    153696   
5     international              6                    269253   
6         beverages              7                   2688115   
7              pets              8                     97716   
8   dry goods pasta              9                    866627   
9              bulk             10                     34573   
10    personal care             11                    447010   
11     meat seafood             12                    708927   
12           pantry             13                   1875369   
13        breakfast             14                    703033   
14     canned goods             15                   1068058   
15       dairy eggs             16                   5398185   
16        household             17                    738666   
17           babies             18                    423794   
18           snacks             19                   2887550   
19             deli             20                   1051249   
20          missing             21                     69145   

    total_spent_in_dept  avg_price_per_item_in_dept  total_orders_from_dept  \
0        17,289,206.000                       7.737                 1180461   
1           253,708.000                       6.991                   35056   
2         9,208,026.000                       7.854                  878658   
3        75,660,936.000                       7.982                 2409320   
4         1,251,654.250                       8.144                   84689   
5         2,068,472.750                       7.682                  221537   
6        20,646,084.000                       7.681                 1456996   
7           770,888.375                       7.889                   59281   
8         6,369,953.500                       7.350                  597862   
9           288,648.594                       8.349                   33802   
10        3,573,172.750                       7.993                  318498   
11       11,551,570.000                      16.294                  574729   
12       15,020,661.000                       8.009                 1117763   
13        5,644,626.000                       8.029                  521401   
14        8,053,132.000                       7.540                  681305   
15       44,954,496.000                       8.328                 2174739   
16        5,454,352.500                       7.384                  470780   
17        3,237,048.500                       7.638                  177710   
18       12,345,538.000                       4.275                 1391447   
19        8,177,352.000                       7.779                  770300   
20          598,845.188                       8.661                   59477   

    avg_items_per_order_by_dept  n_unique_items_in_dept _merge  
0                         1.893                    4006   both  
1                         1.035                     548   both  
2                         1.334                    1515   both  
3                         3.934                    1684   both  
4                         1.815                    1054   both  
5                         1.215                    1139   both  
6                         1.845                    4360   both  
7                         1.648                     971   both  
8                         1.450                    1858   both  
9                         1.023                      38   both  
10                        1.403                    6559   both  
11                        1.233                     906   both  
12              

In [58]:
dept_data['_merge'].value_counts()

both          21
left_only      0
right_only     0
Name: _merge, dtype: int64

In [59]:
# Removing the merge flag so it doesn't interfere with future merges
dept_data = dept_data.drop(columns = ['_merge'])

In [60]:
dept_data.head()

department  department_id  total_items_sold_by_dept  total_spent_in_dept  \
0     frozen              1                   2234743       17,289,206.000   
1      other              2                     36291          253,708.000   
2     bakery              3                   1172428        9,208,026.000   
3    produce              4                   9479291       75,660,936.000   
4    alcohol              5                    153696        1,251,654.250   

   avg_price_per_item_in_dept  total_orders_from_dept  \
0                       7.737                 1180461   
1                       6.991                   35056   
2                       7.854                  878658   
3                       7.982                 2409320   
4                       8.144                   84689   

   avg_items_per_order_by_dept  n_unique_items_in_dept  
0                        1.893                    4006  
1                        1.035                     548  
2                        1.334                    1515  
3                        3.934                    1684  
4                        1.815                    1054

In [61]:
dept_data.shape

(21, 8)

In [62]:
# Reclaiming space
del temp

### 0.5 Confirming the final df

In [68]:
# Confirming final df details
dept_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21 entries, 0 to 20
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   department                   21 non-null     object 
 1   department_id                21 non-null     int64  
 2   total_items_sold_by_dept     21 non-null     int64  
 3   total_spent_in_dept          21 non-null     float32
 4   avg_price_per_item_in_dept   21 non-null     float64
 5   total_orders_from_dept       21 non-null     int64  
 6   avg_items_per_order_by_dept  21 non-null     float64
 7   n_unique_items_in_dept       21 non-null     int64  
dtypes: float32(1), float64(2), int64(4), object(1)
memory usage: 1.4+ KB


In [72]:
# Adjust datatypes for new columns to be more appropriately sized
dept_data['department'] = dept_data['department'].astype('category')
dept_data['department_id'] = dept_data['department_id'].astype('int8')
dept_data['avg_price_per_item_in_dept'] = dept_data['avg_price_per_item_in_dept'].astype('float32')
dept_data['total_orders_from_dept'] = dept_data['total_orders_from_dept'].astype('int32')
dept_data['avg_items_per_order_by_dept'] = dept_data['avg_items_per_order_by_dept'].astype('float32')
dept_data['n_unique_items_in_dept'] = dept_data['n_unique_items_in_dept'].astype('int16')

In [73]:
# Confirming final df details
dept_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21 entries, 0 to 20
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype   
---  ------                       --------------  -----   
 0   department                   21 non-null     category
 1   department_id                21 non-null     int8    
 2   total_items_sold_by_dept     21 non-null     int64   
 3   total_spent_in_dept          21 non-null     float32 
 4   avg_price_per_item_in_dept   21 non-null     float32 
 5   total_orders_from_dept       21 non-null     int32   
 6   avg_items_per_order_by_dept  21 non-null     float32 
 7   n_unique_items_in_dept       21 non-null     int16   
dtypes: category(1), float32(3), int16(1), int32(1), int64(1), int8(1)
memory usage: 1.4 KB


### 0.6 Exporting the new df

In [74]:
#Exporting final df
dept_data.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'department_data.pkl'))